modules/linear_boltzmann_solvers/lbs_solver/lbs_solver.cc

```diff
@@ -742,6 +702,18 @@ LBSSolver::SetOptions(const InputParameters& params)
       }
     }
   } // for p
+
+  if (options_.write_restart_time_interval > 0)
+  {
+    auto dir = options_.write_restart_path.parent_path();
+    {
+      if (opensn::mpi_comm.rank() == 0)
+        std::filesystem::create_directory(dir);
+      opensn::mpi_comm.barrier();
+      if (not std::filesystem::is_directory(dir))
+        throw std::runtime_error("Failed to create restart directory " + dir.string());
+    }
```

Comment:
Does this block need to be in a separate scope?

Reply:
By separate scope, do you mean moved it's own method, or just moving it to a separate scope in-place? I could see moving it to it's own method, but it's fairly small and likely only needed here.

Reply:
Lines 710-714 are within curly braces that are not part of an if statement or anything of the sorts. That's what I am referring to. I forget what the correct terminology for that is.

Reply:
That is the correct terminology, and I just missed it looking at the excerpt above. Fixed!  Thanks!